streaming-binlog/src/main/java/org/openmrs/analytics/DebeziumListener.java

```diff
@@ -0,0 +1,132 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.openmrs.analytics;
+
+import io.debezium.data.Envelope;
+import org.apache.camel.*;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.debezium.DebeziumConstants;
+import org.apache.camel.model.dataformat.JsonLibrary;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+
+import static org.openmrs.analytics.PipelineConfig.getDebeziumConfig;
+
+/**
+ * Debezium change data capture / Listener
+ */
+public class DebeziumListener  extends RouteBuilder {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DebeziumListener.class);
+    private static final String FHIR_ROUTE_HANDLER = "direct:get-fhir";
+
+    // FhirStore
+    private static final FhirStoreUtil FHIR_STORE_UTIL  = new FhirStoreUtil(System.getProperty("cloud.gcpFhirStore","gcp"));
+
+    @Override
+    public void configure() throws Exception {
```

Comment:
A general note about using this Camel RouteBuilder approach: It is great that you have tried the direct debezium embedded approach as well [here](https://github.com/kimaina/openmrs-fhir-analytics/commit/6c1c33d0662ed8dbea75dd2d31eca4424970a02a) and made a comparison. I feel we should document somewhere some of the reasons you mentioned on Slack for preferring this approach over the other. FWIW I found this approach much harder to read and understand but that might be because of my unfamiliarity with Camel.

Reply:
I will document this on WIKI. About readability, DSL language has not always been my favorite, however, I think CAMEL's flexibility outweighs the other option.